sdk-core/src/main/java/care/data4life/sdk/CryptoService.kt

```diff
@@ -0,0 +1,323 @@
+            rng: SecureRandom,
+            base64: Base64,
+            keyFactory: KeyFactory,
+            commonKeyService: CommonKeyService
+    ) {
+        this.alias = alias
+        this.storage = storage
+        this.moshi = moshi
+        this.rng = rng
+        this.base64 = base64
+        this.keyFactory = keyFactory
+        this.commonKeyService = commonKeyService
+    }
+
+    constructor(
+            alias: String,
+            storage: CryptoSecureStore
+    ) {
+        this.alias = alias
+        this.storage = storage
+        moshi = Moshi.Builder().build()
+        rng = SecureRandom()
+        base64 = Base64
+        keyFactory = KeyFactory(base64)
+        commonKeyService = CommonKeyService(alias, storage, keyFactory)
+    }
+
+    fun encrypt(key: GCKey, data: ByteArray): Single<ByteArray> {
+        return Single
+                .fromCallable { data }
+                .map { dataArray ->
+                    val iv = ByteArray(IV_SIZE)
+                    rng.nextBytes(iv)
+                    val ciphertext = symEncrypt(key, dataArray, iv)
+                    val result = ByteArray(iv.size + ciphertext.size)
+                    System.arraycopy(iv, 0, result, 0, iv.size)
+                    System.arraycopy(ciphertext, 0, result, iv.size, ciphertext.size)
+                    result
+                }
+                .onErrorResumeNext { error ->
+                    Log.error(error, "Failed to encrypt data")
+                    Single.error(CryptoException.EncryptionFailed("Failed to encrypt data") as D4LException)
+                }
+    }
+
+    fun decrypt(key: GCKey, data: ByteArray): Single<ByteArray> {
+        return Single
+                .fromCallable { data }
+                .map { dataArray ->
+                    val iv = dataArray.copyOfRange(0, IV_SIZE)
+                    val cipher = dataArray.copyOfRange(IV_SIZE, data.size)
+                    symDecrypt(key, cipher, iv)
+                }
+                .onErrorResumeNext { error ->
+                    Log.error(error, "Failed to decrypt data")
+                    Single.error(CryptoException.DecryptionFailed("Failed to decrypt data") as D4LException)
+                }
+    }
+
+    fun encryptString(key: GCKey, data: String): Single<String> {
+        return encrypt(key, data.toByteArray())
+                .map { dataArray -> base64.encodeToString(dataArray) }
+                .onErrorResumeNext { error ->
+                    Log.error(error, "Failed to encrypt string")
+                    Single.error(CryptoException.EncryptionFailed("Failed to encrypt string") as D4LException)
+                }
+    }
+
+    fun decryptString(key: GCKey, dataBase64: String): Single<String> {
+        return Single
+                .fromCallable { base64.decode(dataBase64) }
+                .flatMap { decoded: ByteArray -> decrypt(key, decoded) }
+                .map { decrypted: ByteArray -> String(decrypted, Charset.forName("UTF-8")) }
```

Comment:
This really discussible: 
"UTF-8" -> StandardCharsets.UTF_8.name()

Reply:
Didn't know that this exists. Found also a nice replacement from Kotlin `Charsets.UTF_8`